src/pani/core.cljs

```diff
@@ -0,0 +1,161 @@
+(ns pani.core
+  (:refer-clojure :exclude [get-in set! reset! conj! swap! deref parents key])
+  (:require cljsjs.firebase
+            [clojure.string :as str]
+            [pani.registry :refer [register-listener]]))
+
+;; constants
+
+(def child-events
+  [:child-added
+   :child-changed
+   :child-moved
+   :child-removed])
+
+(def all-events
+  (conj child-events :value))
+
+;; utils
+
+;; FIXME: camel-case keys?
+;;        hydrate to custom vectors to preserve rich keys?
+;;        preserve sets (don't coerce to vector)
+;;        similarly preserve keywords as values
```

Comment:
@crisptrutski since this are values we put into the firebase on our own (vs `child-events`) I think that should not be a library concern.